scipy/stats/tests/test_device_dtype.py

```diff
@@ -0,0 +1,240 @@
+@pytest.mark.parametrize('dtype', dtypes)
+def test_differential_entropy(method, dtype, xp, devices):
+    if is_array_api_strict(xp) and method == 'correa':
+        pytest.skip("Can't use elipses and integer indices")
+    if is_dask(xp) and method in {'ebrahimi', 'correa', 'auto'}:
+        pytest.skip("method not compatible with Dask")
+
+    dtype = getattr(xp, dtype)
+    for device in devices:
+        values = get_arrays(1, device=device, dtype=dtype, xp=xp)[0]
+        res = stats.differential_entropy(values, method=method)
+        assert xp_device(res) == xp_device(values)
+        assert values.dtype == dtype
+
+
+@pytest.mark.skip_xp_backends('dask.array', reason='no take_along_axis')
+@pytest.mark.parametrize('method', ['inverted_cdf', 'averaged_inverted_cdf',
+                                    'closest_observation', 'interpolated_inverted_cdf',
+                                    'hazen', 'weibull', 'linear', 'median_unbiased',
+                                    'normal_unbiased', 'harrell-davis'])
+@pytest.mark.parametrize('dtype', dtypes)
+def test_quantile(method, dtype, xp, devices):
+    if (is_array_api_strict(xp) or is_torch(xp)) and method == 'harrell-davis':
+        pytest.skip("'harrell-davis' not currently not supported on GPU.")
+
+    dtype = getattr(xp, dtype)
+    for device in devices:
+        values = get_arrays(1, device=device, dtype=dtype, xp=xp)[0]
+        res = stats.quantile(values, 0.5, method=method)
+        assert xp_device(res) == xp_device(values)
+        assert values.dtype == dtype
+
+
+@pytest.mark.parametrize('dtype', dtypes)
+def test_boxcox_llf(dtype, xp, devices):
+    dtype = getattr(xp, dtype)
+    for device in devices:
+        data = get_arrays(1, device=device, dtype=dtype, xp=xp)[0]
+        res = stats.boxcox_llf(1, data)
+        assert xp_device(res) == xp_device(data)
+        assert data.dtype == dtype
+
+
+@pytest.mark.parametrize('fun, kwargs',
+    [(stats.moment, {'order': 2}),
+     (stats.skew, {}),
+     (stats.skew, {'bias': False}),
+     (stats.kurtosis, {}),
+     (stats.kurtosis, {'bias': False}),
+     (stats.sem, {}),
+     (stats.kstat, {'n': 1}),
+     (stats.kstat, {'n': 2}),
+     (stats.kstat, {'n': 3}),
+     (stats.kstat, {'n': 4}),
+     (stats.kstatvar, {'n': 1}),
+     (stats.kstatvar, {'n': 2}),
+     (stats.circmean, {}),
+     (stats.circvar, {}),
+     (stats.circstd, {}),
+     (_xp_var, {}),
+     (stats.gstd, {}),
+     (stats.variation, {}),
+     (stats.tmean, {'limits': (0.1, 0.9)}),
+     (stats.tvar, {'limits': (0.1, 0.9)}),
+     (stats.tmin, {'lowerlimit': 0.1}),
+     (stats.tmax, {'upperlimit': 0.9}),
+     (stats.tstd, {'limits': (0.1, 0.9)}),
+     (stats.tsem, {'limits': (0.1, 0.9)}),
+     ])
+@pytest.mark.parametrize('dtype', dtypes)
+def test_one_in_one_out(fun, kwargs, dtype, xp, devices):
+    if is_dask(xp) and fun == stats.variation:
+        pytest.skip("dtype inference failing in xpx.apply_where")
```

Comment:
@mdhaber this is quite worrysome:
> E           TypeError("Multiple namespaces for array inputs: {<module 'scipy._lib.array_api_compat.numpy' [...], <module 'scipy._lib.array_api_compat.dask.array' [...]

Reply:
I'm going to look into it, but it wasn't worth holding up the whole PR. Dask can be very creative about how to fail with code that works for other backends.

Reply:
> Dask can be very creative about how to fail with code that works for other backends.

Yeah I looked at this, and it's very much not obvious what is going on there. All the input variables seemed to be Dask arrays of the same dtype and device, and internally somewhere, one is converted to a numpy array.